app/src/test/java/illyan/jay/view_model/LoginViewModelTest.kt

```diff
@@ -0,0 +1,205 @@
+package illyan.jay.view_model
+
+import android.app.Activity
+import co.zsmb.rainbowcake.test.assertObserved
+import co.zsmb.rainbowcake.test.base.ViewModelTest
+import co.zsmb.rainbowcake.test.observeStateAndEvents
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+import illyan.jay.ui.login.*
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Test
+import org.junit.Before
+import timber.log.Timber
+
+@ExperimentalCoroutinesApi
+class LoginViewModelTest: ViewModelTest() {
+    private lateinit var mockPresenter: LoginPresenter
+    private lateinit var viewModel: LoginViewModel
+    private lateinit var mockTree: Timber.Tree
+
+    @Before
+    fun initEach() {
+        mockPresenter = mockk()
+        viewModel = LoginViewModel(mockPresenter)
+        mockTree = mockk()
+        Timber.plant(mockTree)
+    }
+
+    @After
+    fun destroyEach() {
+        Timber.uproot(mockTree)
+    }
+
+    @Test
+    fun refresh_sets_state_to_logged_in_when_user_is_logged_in() {
```

Comment:
Ez a sor így is kinézhetne: ``fun `Refresh sets state to logged in when user is logged in`() {``. Ez kihasználja a kotlin adottságait és olvashatóbb szerintem.

Reply:
Javítva. Durva, hogy ilyet lehet csinálni kotlinban. Ugye ez sima függvényekre nem működik? :D

Reply:
Azt hiszem ott is lehet :D

Reply:
Na jó. :D Ez azért nekem már túl egzotikus.